xds/internal/client/bootstrap/bootstrap.go

```diff
@@ -160,13 +160,19 @@ func bootstrapConfigFromEnvVariable() ([]byte, error) {
 // fields left unspecified, in which case the caller should use some sane
 // defaults.
 func NewConfig() (*Config, error) {
-	config := &Config{}
-
 	data, err := bootstrapConfigFromEnvVariable()
 	if err != nil {
 		return nil, fmt.Errorf("xds: Failed to read bootstrap config: %v", err)
 	}
 	logger.Debugf("Bootstrap content: %s", data)
+	return NewConfigFromContents(data)
+}
+
+// NewConfigFromContents returns a new Config using the specified bootstrap
+// file contents instead of reading the environment variable.  This is only
+// suitable for testing purposes.
+func NewConfigFromContents(data []byte) (*Config, error) {
```

Comment:
Same here: `NewConfigFromContentsForTesting`?

Reply:
It doesn't seem worthwhile to use this naming given:

1. it's internal-only, and
2. if we ever had a desire to call this from our production implementation, we'd probably just do it and remove the `FromTesting` at that time.